### PR TITLE
[GPU] GroupNormalization opt kernels

### PR DESCRIPTION
**Pull request by e-ddykim**
### Details:
 - This PR adds two group_normalization opt kernels, group_normalization_bfyx and group_normalization_b_fs_yx_fsv16.

### Tickets:
 - 144094

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp [hunk 1]
```diff
@@ -294,6 +296,8 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
                 return !is_type<ov::op::v0::MatMul>(next_node);
             });
 
+        manager.register_pass<ov::intel_gpu::GroupNormComposition>();
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp:299]: GroupNormDecomposition is being disabled, but still need Composition too?

**Review by yeonbok — APPROVED** [review]

File: src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp (hunk 1 above)
Comment (e-ddykim) [reply src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp:299]: Current OV stable diffusion models does not have `group_normalization` layers. They have decomposed group_normalization layers instead. So we need a composition pass to apply `group_normalization` in the current OV SD models.

File: src/plugins/intel_gpu/src/plugin/transformations/group_norm_composition.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+#include "openvino/op/group_normalization.hpp"
+#include "openvino/op/squeeze.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "transformations/utils/utils.hpp"
+
+namespace ov {
+namespace intel_gpu {
+
+GroupNormComposition::GroupNormComposition() {
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/plugin/transformations/group_norm_composition.cpp:24]: 1. That can be a common pass.
2. Are new IRs generated w/ decomposed GroupNorm?
3. Rename to `GroupNormFusion` to be aligned with other passes

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/plugin/transformations/group_norm_composition.cpp:24]: 1. Okay, I'll do it in a separate PR.
2. Yes, new IRs have decomposed GroupNorm layers.
3. Okay, I'll also apply it with #1.

**Pull request merged**